### PR TITLE
Updated plugin function return types to be Nodes rather than file name strings.

### PR DESCRIPTION
**Pull request by ghost**
It seems far more useful and flexible for plugins to access the actual instanced scene objects that are opened, than to get strings containing their file names.  It doesn't seem intentional, but I wouldn't know the use case exactly.

I changed get_open_scenes() to return the Nodes, which is what the name implies.

Updated the closing scene signal to emit a reference to the Node that is closing.  It should be fine to use from my testing, as long as it gets used right away, since the delete happens after the plugins are all signaled.

godot windows opt tools 64_2018-09-12_13-34-30

I updated the documentation, the comment about "freshly opened" scenes was not entirely accurate.

With this change if you require the file names, it's easy to access them with `node.filename`, but when having only a file name string, it is very difficult to get back the actual opened scene object in the editor.

It could be a small break in compatibility, but how many people are using it specifically for tracking filenames of editor scenes?

### CODE REVIEW

File: doc/classes/EditorPlugin.xml [hunk 1]
```diff
@@ -454,14 +454,14 @@
 			<argument index="0" name="scene_root" type="Node">
 			</argument>
 			<description>
-				Emitted when user change scene. The argument is a root node of freshly opened scene.
+				Emitted when the user changes editor scenes.  Either from a tab selection or opening a new scene. The argument will contain the root node of the scene that has become active.  (Changing to empty scenes will not emit this signal.)
```
[the remark this answers is not in the document]
Comment (avencherus) [reply doc/classes/EditorPlugin.xml]: Thanks, missed it.  Haven't gotten properly versed yet with the docs.

**Comment by avencherus** [comment]
@akien-mga Fixed up the XML and amended the commit.

**Comment by reduz** [comment]
I have two concerns with this

* Does it break current plugins? if so this may have to wait for 4.0
* I think it may be possible that sometimes scenes are re-packed and instanced when switching, so node pointer may break. This needs to be tested

**Comment by avencherus** [comment]
I do wonder if anyone is presently using it for a list of file names.  The only reason I bumped into it was that I was expected to have an easy way to get references to the open scenes, but got back an array of strings instead.  All I could immediately do with that, that was useful, is just check the size to see if none were open.

I at least could see that scenes are held in memory, but pulled from the scene tree when switched away from.  So the only way to get a list of scenes with a plugin is to keep a collection of them as you see them appear from the scene_changed() signal or scan for differences in the viewport they get parented to.  Then periodically at some point probe your array for empty references and clean them up.

In regards to the pointer issues I'm afraid I'm not terribly confident or experienced there.  At least on the editor side it looks like the references held up while changing scenes.  But I imagine you mean something a bit deeper.

[Open Scenes Reference Test.zip](https://github.com/godotengine/godot/files/2376489/Open.Scenes.Reference.Test.zip)

**Comment by avencherus** [comment]
There may be potential issues with scene references when a scene is renamed inside the editor.  I don't have the familiarity with the feature or the time to investigate.  To be on the safe side, I'm withdrawing this.

**Pull request closed without merging**